src/Compilers/VisualBasic/Portable/Symbols/Source/SynthesizedFieldSymbol.vb

```diff
@@ -55,7 +55,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             End Get
         End Property
 
-        Friend Overrides Sub AddSynthesizedAttributes(moduleBuilder As PEModuleBuilder, ByRef attributes As ArrayBuilder(Of SynthesizedAttributeData))
+        Friend Overrides Sub AddSynthesizedAttributes(moduleBuilder As PEModuleBuilder, ByRef attributes As ArrayBuilder(Of VisualBasicAttributeData))
+            If TypeOf Me Is LambdaCapturedVariable OrElse TypeOf Me Is StateMachineFieldSymbol Then
+                Dim definition = TryCast(_implicitlyDefinedBy.OriginalDefinition, SourceParameterSymbolBase)
+
+                If definition IsNot Nothing AndAlso
+                   ContainingModule = definition.ContainingModule Then
+                    For Each attr In definition.GetAttributes()
+                        Dim attributeType As NamedTypeSymbol = attr.AttributeClass
+                        If attributeType IsNot Nothing AndAlso attributeType.HasCompilerLoweringPreserveAttribute AndAlso
+                           (attributeType.GetAttributeUsageInfo().ValidTargets And System.AttributeTargets.Field) <> 0 Then
+                            AddSynthesizedAttribute(attributes, attr)
+                        End If
+                    Next
```

Comment:
This block is used in a couple of locations. Would it make sense to extract a helper method?